packages/strapi-plugin-graphql/services/GraphQL.js

```diff
@@ -17,6 +17,424 @@ const GraphQLJSON = require('graphql-type-json');
+            );
+            return strapi.utils.models.convertParams(name, params);
+          }
+        },
+        [connectionGlobalId]: {
+          values: (obj, option, context) => {
+            // Object here contains the key/value of the field that has been grouped-by
+            // for instance obj = { where: { country: 'USA' } } so the values here needs to be filtered according to the parent value
+            return modelResolver(obj, obj, context);
+          },
+          groupBy: (obj, option, context) => { // eslint-disable-line no-unused-vars
+            // There is noting to resolve here, it's the aggregation resolver that will take care of it
+            return obj;
+          },
+          aggregate: (obj, option, context) => { // eslint-disable-line no-unused-vars
+            // There is noting to resolve here, it's the aggregation resolver that will take care of it
+            return obj;
+          },
+        },
+        ...aggregatorFormat.resolver,
+        ...groupByFormat.resolver,
+      },
+    };
+  },
+
+  /**
+   * Generate the connection type of each non-array field of the model
+   * 
+   * @return {String}
+   */
+  generateConnectionFieldsTypes: function (fields, model) {
+    const { globalId } = model;
+
+    const primitiveFields = this.getFieldsByTypes(
+      fields,
+      this.isNotOfTypeArray,
+      type => this.extractType(type),
+    );
+
+    const connectionFields = _.mapValues(primitiveFields, (fieldType) => ({
+      key: fieldType,
+      connection: `${globalId}Connection`,
+    }));
+
+    let connectionFieldsTypes = '';
+    _.forEach(primitiveFields, (fieldValue, fieldKey) => {
+      connectionFieldsTypes += `type ${globalId}Connection${_.upperFirst(fieldKey)} {${this.formatGQL(connectionFields[fieldKey])}}\n\n`;
+    });
+
+    return connectionFieldsTypes;
+  },
+
+  formatConnectionGroupBy: function(fields, model, name) {
+    const { globalId } = model;
+    const groupByGlobalId = `${globalId}GroupBy`;
+
+    // Extract all primitive fields and change their types
+    const groupByFields = this.getFieldsByTypes(
+      fields,
+      this.isNotOfTypeArray,
+      (fieldType, fieldName) => `[${globalId}Connection${_.upperFirst(fieldName)}]`,
+    );
+
+    // Get the generated field types
+    let groupByTypes = `type ${groupByGlobalId} {${this.formatGQL(groupByFields)}}\n\n`;
+    groupByTypes += this.generateConnectionFieldsTypes(fields, model);
+
+    return {
+      globalId: groupByGlobalId,
+      type: groupByTypes,
+      resolver: {
+        [groupByGlobalId]: this.createGroupByFieldsResolver(model, groupByFields, name),
+      }
+    };
+  },
+
+  formatConnectionAggregator: function(fields, model) {
+    const { globalId } = model;
+    
+    // Extract all fields of type Integer and Float and change their type to Float
+    const numericFields = this.getFieldsByTypes(fields, (fieldType) => {
+      return fieldType === 'Int' || fieldType === 'Float';
```

Comment:
I see a lot of 
```js
(fieldType) => {
 return fieldType === 'Int' || fieldType === 'Float'
}
```

it can be change by:
```js
fieldType => fieldType === 'Int' || fieldType === 'Float'
```

Moreover, it's used several times around line 415, so it can be factoring into a function as you make for`isPrimitiveType`, with a name as `isNumberType` or something else.